### ISSUE TITLE
Icicles minibuffer bindings no longer work...

### ISSUE THREAD

**Issue by mattfidler** [ergoemacs/ergoemacs-mode#406]
Also I've found another icicles-specific thing that does not work anymore. [Those minibuffer bindings.](https://github.com/ergoemacs/ergoemacs-mode/blob/3530a11d796cc92aec316d9160b3de4fcdc74cd4/ergoemacs-themes.el#L483) I was able to fix it in my config like this:
```lisp
(ergoemacs-theme-component icicles-minibuffer ()
  ""
  (add-hook
   'icicle-minibuffer-setup-hook
   '(lambda ()
      (define-key minibuffer-local-map (kbd "<f11>") 'previous-history-element)
      (define-key minibuffer-local-map (kbd "M-<f11>") 'icicle-insert-history-element)
      (define-key minibuffer-local-map (kbd "<f12>") 'next-history-element)
      (define-key minibuffer-local-map (kbd "S-<f11>") 'next-history-element)
      (define-key minibuffer-local-map (kbd "M-<f12>") 'icicle-insert-history-element)
      (define-key minibuffer-local-map (kbd "S-<f12>") 'previous-history-element))))
(ergoemacs-require 'icicles-minibuffer)
```
See Issue #372 @sg2002

**Comment by mattfidler** [comment]
This should be fixde @sg2002

**Comment by sg2002** [comment]
There's still an issue here. 

When you start emacs anew, doing M-a and then cycling history using f11 and f12 works. But only the first time. After the first time, it stops working. 

Cycling C-x b stops working too. But it never breaks by itself, until when you do M-a.

Cycling C-o works all the time.

The keys get interpreted as previous-line and next-line.

**Comment by mattfidler** [comment]
I think fixing issue #407 broke this...

**Comment by mattfidler** [comment]
This should be fixed now (at least it works for me in a basic ergoemacs+icy setup).

**Comment by sg2002** [comment]
Yeah, it works. I've also noticed another problem with icicles minibuffer bindings.

S-TAB in ergoemacs now always runs icicle-complete-keys. This should not happen during minibuffer completion. Instead icicle-apropos-complete should be ran.

**Comment by sg2002** [comment]
Ok, I've solved that. There's a variable called icicle-key-complete-keys-for-minibuffer. In my custom it was set to nil. Probably leftover from some testing. Ergoemacs overrides this variable anyway. But basically, if you have it set to nil at the moment ergoemacs loads, icicle-complete-keys starts overriding icicle-apropos-complete.

This happens since ef27870f679db38b0297e108b39547818fa9075b .

Probably does not require any action, but good to keep in mind in case anyone starts having similar problems.

**Comment by mattfidler** [comment]
Thanks @sg2002

**Comment by sg2002** [comment]
Another really minor thing. M-o is currently not overridden in icicles minibuffer. I was able to fix it by adding `(define-key minibuffer-local-map (kbd "M-o") 'forward-word)` to the same block.

**Comment by mattfidler** [comment]
Thanks.  I will look into it.

**Comment by sg2002** [comment]
Ok, another one, in minibuffer with icicles, M-s and M-r get treated as the vanilla next-matching-history-element and previous-matching-history-element.

**Comment by mattfidler** [comment]
This tests the f11 and f12 bindings: 6c5d5bc7a4e366dec328dfd3430f20b7b977f781

**Comment by mattfidler** [comment]
What do you expect the `M-s` and M-r` keys to do?

**Comment by mattfidler** [comment]
For f11 f12 M-o M-s M-r I get the following commands: `(previous-history-element next-history-element icicle-insert-history-element ergoemacs-move-cursor-next-pane previous-matching-history-element)`

**Comment by sg2002** [comment]
I expect M-s to be ergoemacs-move-cursor-next-pane. And M-r to have it's normal ergoemacs kill-word binding.

Icy M-o is already rebound in ergoemacs to M-f11 and M-f12.

**Comment by mattfidler** [comment]
When I run the test `M-s` is `ergoemacs-move-cursor-next-pane`.  `M-r` is previous-matching-history-element`.

**Comment by mattfidler** [comment]
Should this be tested in command other than `icicle-execute-extended-command`?

**Comment by sg2002** [comment]
No. I've found the conditions under which this happens. First the config:
```
(require 'icicles)
(icy-mode 1)
(require 'ergoemacs-mode)
(ergoemacs-mode 1)
(ergoemacs-theme-component reclaim-C-f ()
  "We need to give at least one sequence to reclaim C-f from isearch and get the new icicle-search-key-prefix picked up."
  (global-set-key (kbd "C-f .") 'isearch-forward-symbol-at-point))
(ergoemacs-require 'reclaim-C-f)
(setq icicle-search-key-prefix (kbd "C-f"))
```
Now the triggers:
1. `(require 'smart-mode-line)(sml/setup)`
2. `(require 'srefactor)`
3. `(require 'virtualenvwrapper)`

If you have any of these in your config after `(setq icicle-search-key-prefix (kbd "C-f"))` then this M-s issue happens.

This thing is also infectious. That is, if you do desktop-save from infected version and then read it in in a clean one, it becomes infected too. When you save desktop in the secondary infected version, the infection does not transfer. This behavior had me totally baffled. ;-)

**Comment by mattfidler** [comment]
Is this only in the minibuffer?

**Comment by sg2002** [comment]
Yes.

**Comment by mattfidler** [comment]
- Smart mode line doesn't change anything in the search map.
- srefactor doesn't change anything in the search map.
- virtualenvwrapper doesn't change anything in the search map.

I'm unsure how any of these packages would affect the ergoemacs-mode state. I'm a bit baffled as well.

**Comment by mattfidler** [comment]
Neither of these packages uses global-set-key either

**Comment by mattfidler** [comment]
Nor do they modify the minibuffer.  Only virtualenvwrapper uses the minibuffer...

**Comment by mattfidler** [comment]
I updated test 407 to include what you described as something that changes `M-s` binding.  However, when I run `(ert "ergoemacs-test-407")`, I cannot reproduce your issue.

**Comment by sg2002** [comment]
Test 407 worked for me too, at first. But replacing `ergoemacs-package`'s with `require`'s, had the desired effect.

**Comment by mattfidler** [comment]
So replacing ergoemacs-package does not cause an issue for you, but using
require does.
